# more guards for boxes where we cross-compile with ruby 2.0.0p598
diff --git a/lib/delegate.rb b/lib/delegate.rb
index 1587c7e..a53690f 100644
--- a/lib/delegate.rb
+++ b/lib/delegate.rb
@@ -107,6 +107,12 @@ class Delegator < BasicObject
   private_constant :KERNEL_RESPOND_TO
 
   # Handle BasicObject instances
+  if ENV['CROSS_COMPILING']
+    def self.private(name, *)
+      # nil
+    end
+  end
+
   private def target_respond_to?(target, m, include_private)
     case target
     when Object
diff --git a/lib/rubygems/core_ext/kernel_warn.rb b/lib/rubygems/core_ext/kernel_warn.rb
index dfc577d..3edf965 100644
--- a/lib/rubygems/core_ext/kernel_warn.rb
+++ b/lib/rubygems/core_ext/kernel_warn.rb
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 # `uplevel` keyword argument of Kernel#warn is available since ruby 2.5.
-if RUBY_VERSION >= "2.5"
+if RUBY_VERSION >= "2.5" && ENV['CROSS_COMPILING'].nil?
 
   module Kernel
     path = "#{__dir__}/" # Frames to be skipped start with this path.
